### PR TITLE
Honor workspace trust in VSCode. Also warn about possible code being executed under --dry-run first time it is invoked.

### PR DESCRIPTION
**Pull request by andreeis**
Honor workspace trust in VSCode.
Also warn about possible code being executed under --dry-run first time it is invoked.
Fix for https://github.com/microsoft/vscode-makefile-tools/issues/506.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -568,12 +568,41 @@
...
                 {
                     "id": "makefile.outline",
+                    "when": "isWorkspaceTrusted",
                     "name": "%makefile-tools.configuration.views.makefile.outline.description%"
                 }
             ]
         },
         "menus": {
             "commandPalette": [
+                {
```
Comment (gcampbell-msft) [inline package.json:578]: Nit: inconsistent spacing.

File: src/make.ts [hunk 2]
```diff
@@ -969,10 +978,46 @@ function getRelevantConfigStats(stats: any): ConfigureSubphaseStatusItem[] {
...
+                   "This is needed to calculate accurate IntelliSense and targets information. " +
+                   "Although in general 'make --dry-run' only lists (without executing) the operations 'make' would do in the current context, " +
+                   "it is still possible some code to be executed, like $(shell) syntax in the makefile or recursive invocations of the $(MAKE) variable.");
+    logger.message("If you don't feel comfortable allowing this configure process and 'make --dry-run' to be invoked by the extension, " +
+                   "you can chose a recent full, clean, verbose and up-to-date build log as an alternative, via the setting 'makefile.buildLog'. ");
+    // Also, show the output channel for that message to be visible.
+    logger.showOutputChannel();
+    let yesButton: string = localize("yes.dont.show.again", "Yes (don't show again)");
+    let noButton: string = localize("no", "No");
+    const chosen: vscode.MessageItem | undefined = await vscode.window.showErrorMessage<vscode.MessageItem>("Configuring project. Code can still execute in --dry-run mode. Do you want to continue?",
```
Comment (gcampbell-msft) [inline src/make.ts:998]: Please make sure to localize the necessary strings. It's not always required to localize things going to the logging output, we should follow the precedent we've set in this extension (I don't think we localize), but the popup should certainly be localized.

**Review by gcampbell-msft — APPROVED** [review]
I haven't tested, but besides a couple of changes for localization and nit changes, this looks good!

File: package.json (hunk 1 above)
Comment (andreeis) [reply package.json:578]: done

**Pull request merged**